route-rs-runtime/src/link/sync_link.rs

```
@@ -44,8 +44,8 @@ impl<E: Element> Stream for SyncLink<E> {
         match input_packet_option {
             None => Ok(Async::Ready(None)),
             Some(input_packet) => {
-                let output_packet: E::Output = self.element.process(input_packet);
-                Ok(Async::Ready(Some(output_packet)))
+                let output_packet_option: Option<E::Output> = self.element.process(input_packet);
```

Same as a above, we don't want to return the None since that is a teardown signal. See line 29 docs.  Instead we should silently drop the None, and pull another packet for processing, until we exhaust the upstream, or have a valid packet to return.

---

You should also update the documentation above here to note that in the Ready(Some) case, the packet is processed by the element, which may or may not yield a valid return.